library/swift/src/filters/Filter.swift

```diff
@@ -4,14 +4,33 @@ import Foundation
 /// Interface representing a filter. See `RequestFilter` and `ResponseFilter` for more details.
 public protocol Filter {
   /// A unique name for a filter implementation. Needed for extension registration.
-  var name: String { get }
+  static var name: String { get }
+
+  /// Required initializer for internal creation.
+  init()
+}
+
+extension Filter {
+  static func create() -> EnvoyHTTPFilter {
+    return EnvoyHTTPFilter(filter: Self.init())
+  }
+}
```

Comment:
I'd probably remove this and just it below using `filterType.init()`

Reply:
Yep, good catch - this is leftover from an earlier approach.